### PR TITLE
New logging implementation

### LINKED ISSUE #692: Adjust logger options to take values of Logger interface
### Description

We also want to deprecate the old `LoggingFunc`, so adding warnings for when that is used is also needed.

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

### PR DESCRIPTION
**Pull request by aoberoi**
###  Summary

Implements `ConsoleLogger`, which allows us to remove `loglevel` dependency.

Refactors objects to use the new `ConsoleLogger`.

Refines the `Logger` interface.

Accept `Logger` objects for the `logger` option on `WebClient`, `RTMClient`, and `KeepAlive`.

Adjusts documentation for deprecation of `LoggingFunc`

Fixes #692 

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: docs/_pages/rtm_client.md [hunk 1]
```diff
@@ -307,10 +307,31 @@ not log anything as long as nothing goes wrong.
 You can adjust the log level by setting the `logLevel` option to any of the values found in the `LogLevel` top-level
```
Comment (shanedewael) [inline docs/_pages/rtm_client.md]: Can we list out the options here?

File: docs/_pages/web_client.md [hunk 2]
```diff
@@ -426,8 +426,31 @@ not log anything as long as nothing goes wrong.
 You can adjust the log level by setting the `logLevel` option to any of the values found in the `LogLevel` top-level
```
Comment (shanedewael) [inline docs/_pages/web_client.md]: Same as above comment

File: docs/_pages/web_client.md [hunk 3]
```diff
@@ -426,8 +426,31 @@ not log anything as long as nothing goes wrong.
...
+|--------------|-------------------|-------------|
+| `setLevel()` | `level: LogLevel` | `void`      |
+| `setName()`  | `name: string`    | `void`      |
+| `debug()`    | `...msgs: any[]`  | `void`      |
+| `info()`     | `...msgs: any[]`  | `void`      |
+| `warn()`     | `...msgs: any[]`  | `void`      |
+| `error()`    | `...msgs: any[]`  | `void`      |
+
+
+**NOTE**: The option can also take a function with the following signature, but this usage is deprecated:
```
Comment (shanedewael) [inline docs/_pages/web_client.md]: Suggested edit: "While the usage of logging functions is deprecated, the `logger` option will still currently accept a function if the method signature matches `fn(level: string, message: string)`."

The main thing I want to change is for the deprecation part to be at the beginning of the note.

**Comment by aoberoi** [comment]
> 89.36% of diff hit (target 91.97%)

i'm going to override this failure because the lines that are not hit are lines that will be likely removed in v5 (they are in the compatibility layer for logging functions).

**Pull request merged**